src/lib/operators/aggregate.cpp

```diff
@@ -334,7 +342,14 @@ void Aggregate::_aggregate() {
         Store unique IDs for equal values in the groupby column (similar to dictionary encoding). 
         The ID 0 is reserved for NULL values. The combined IDs build an AggregateKey for each row.
         */
-        auto id_map = std::unordered_map<ColumnDataType, AggregateKeyEntry>();
```

Comment:
Can we estimate the number of key elements here? We could pre-size the hash map accordingly.
Resizing and rehashing is expensive.

Reply:
Hmm, there are a lot of places where that would help, but rather than hacking it in everywhere, I believe that this is an information that should come from a central statistics component, don't you think?

Reply:
I would not call it "hacking" it but call it a safe performance assumption. :)
Similar to the vector pre-sizing we might win some performance for minimal added costs in memory.

But I have to admit that I have not fully read the code. Is there a chance that we might know roughly how large (thousands, millions?) the table is going to be?

Reply:
`decreased by .21 compared to master and is now 85.47%.` 😠

Reply:
Probably because it is not based on the currecnt master.

> But I have to admit that I have not fully read the code. Is there a chance that we might know roughly how large (thousands, millions?) the table is going to be?

It all depends on the value distribution. As other operators (e.g., table scan) also would profit from knowing the distribution, I don't think we should implement sampling right in the operators.

Reply:
I agree that sampling should not be a part of the operator. I was just wondering if we could have a better estimate of the output size at this time during execution.
If not, that's all fine.